### PR TITLE
explicit 8-byte alignment

### PR DESCRIPTION
**Pull request by cavemanloverboy**
all padding fields are now 8 byte aligned (on bpf, arm, x86), and all program types are now explciitly 8 byte aligned.

improves devex on mac. bpf in-memory (on-chain) representation is unchanged.

### CODE REVIEW

**Review by exdx — COMMENTED** [review]
Does this change make it possible to finally compile marginfi programs on M-series macs? That would be a huge plus and remove the need for having a x86 VM to compile for me. 

I'm a big fan of this change but would like @jkbpvsc to make the final approval. This code is very sensitive and if we change it in a way that's not backwards compatible it will be a pain.

**Comment by cavemanloverboy** [comment]
yes it makes it compatible with mac. i am confident it doesn't affect on-chain representation, as it just makes the alignment and size of types equal between bpf and arm without changing bpf, but someone should definitely write a test that fetches all types from mainnet to be doubly sure.

**Review by jkbpvsc — APPROVED** [review]

**Comment by jkbpvsc** [comment]
@cavemanloverboy can you fix linting, it seems its failing

**Comment by cavemanloverboy** [comment]
> @cavemanloverboy can you fix linting, it seems its failing

this failure was on main haha. i'll fix

**Comment by cavemanloverboy** [comment]
the linter failed because the bank and oracles are no longer in the vector holding the lut keys. 

```
    // keys.extend(bank_pks);
    // keys.extend(oracle_pks);
```

Just wanna confirm this is intentional

**Comment by losman0s** [comment]
I added a couple of fixture-based tests on `Bank`'s and `MarginfiAccount`'s `WrappedI80F48` fields (most of them at least). I took the values with the current main, so that it reflects what is deployed rn.

You can rebase on `main` and run those to have that additional confirmation that layout's still ok.

**Review by exdx — APPROVED** [review]

**Comment by exdx** [comment]
Closes #181

File: programs/marginfi/src/macros.rs [hunk 1]
```diff
@@ -100,3 +100,10 @@ macro_rules! assert_struct_size {
         static_assertions::const_assert_eq!(std::mem::size_of::<$struct>(), $size);
     };
 }
+
+#[macro_export]
+macro_rules! assert_struct_align {
+    ($struct: ty, $align: expr) => {
+        static_assertions::const_assert_eq!(std::mem::align_of::<$struct>(), $align);
```
[the remark this answers is not in the document]
Comment (cavemanloverboy) [reply programs/marginfi/src/macros.rs:107]: what do you want to check exactly?

**Comment by cavemanloverboy** [comment]
^ last commit, just wanted to commit latest lock

**Pull request merged**